File: week2/extraction_demo.py
"""Runs Exercise 2's extraction logic directly against the sample emails in config.py,
without needing the FastAPI server running - useful for a quick demo or for testing the
reject/retry logic in isolation. Run: python extraction_demo.py
"""

import asyncio

import config
import extraction
import utils


async def main():
    for email in config.SAMPLE_SUPPORT_EMAILS:
        utils.print_panel("Input email", email)
        result = await extraction.extract_support_ticket(email)
        if result["error"]:
            utils.print_panel("Extraction FAILED", result["error"])
        else:
            fields = "\n".join(f"{key}: {value}" for key, value in result["ticket"].items())
            utils.print_panel(f"Extracted ticket (succeeded on attempt {result['attempts']})", fields)


if __name__ == "__main__":
    asyncio.run(main())

File: week2-js/src/extractionDemo.ts
/**
 * Runs Exercise 2's extraction logic directly against the sample emails in config.ts,
 * without needing the Express server running - useful for a quick demo or for testing
 * the reject/retry logic in isolation. Run: npm run extraction-demo
 */

import * as config from "./config.js";
import * as extraction from "./extraction.js";
import { printPanel } from "./utils.js";

async function main(): Promise<void> {
  for (const email of config.SAMPLE_SUPPORT_EMAILS) {
    printPanel("Input email", email);
    const result = await extraction.extractSupportTicket(email);
    if (result.error) {
      printPanel("Extraction FAILED", result.error);
    } else {
      const fields = Object.entries(result.ticket!)
        .map(([key, value]) => `${key}: ${value}`)
        .join("\n");
      printPanel(`Extracted ticket (succeeded on attempt ${result.attempts})`, fields);
    }
  }
}

main();
